sdk-tools/src/sign.rs

```diff
@@ -0,0 +1,272 @@
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
+
+    /// Generate the command to create the data required for offline signing,
+    /// and write it to the given output data path.
+    pub fn gendata(
+        &mut self,
+        unsigned_enclave: &Path,
+        config_path: &Path,
+        output_datfile: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("gendata")
+            .arg("-enclave")
+            .arg(unsigned_enclave)
+            .arg("-config")
+            .arg(config_path)
+            .arg("-out")
+            .arg(output_datfile);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
+
+    /// Combine an unsigned enclave and signature into the output enclave, after
+    /// checking the signature.
+    pub fn catsig(
+        &mut self,
+        unsigned_enclave: &Path,
+        config_path: &Path,
+        public_key_pem: &Path,
+        gendata_output: &Path,
+        signature: &Path,
+        output_enclave: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("catsig")
+            .arg("-enclave")
+            .arg(unsigned_enclave)
+            .arg("-config")
+            .arg(config_path)
+            .arg("-key")
+            .arg(public_key_pem)
+            .arg("-unsigned")
+            .arg(gendata_output)
+            .arg("-sig")
+            .arg(signature)
+            .arg("-out")
+            .arg(output_enclave);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        cmd
+    }
+
+    /// Examine a signed enclave file and dump the data
+    pub fn dump(
+        &mut self,
+        signed_enclave: &Path,
+        css_file_path: &Path,
+        dump_file_path: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("dump")
+            .arg("-enclave")
+            .arg(signed_enclave)
+            .arg("-dumpfile")
+            .arg(dump_file_path)
+            .arg("-cssfile")
+            .arg(css_file_path);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
```

Comment:
:thought_balloon: See comment in the `From` I'm thinking this method should go away and come back in a struct, in a subsequent PR, whose only porpoise is to dump signed enclave contents

Reply:
Will be fixed in https://github.com/mobilecoinfoundation/sgx/issues/264